Security/Risk_management/Rapid_Risk_Assessment.mediawiki

```diff
@@ -0,0 +1,362 @@
+* '''Easy to update'''. Can be run during the architecture phase, and re-run if the project is re-architected or new high level design decisions are made.
+* '''Informative''' It collects risk impact, an approximate impact likelihood and data dictionary.
+
+This helps to make the following type of risk-based decisions:
+
+* Is the security provided by a given platform appropriate to host a specific type of data?
+* How much should we care about maintenance, etc?
+* Does this service warrant a full security threat-model, peneration test, privacy review?
+* How long should we spend on these tasks?
+* Is there anything obvious I should really look at fixing right now?
+
+{| class="wikitable"
+|-
+! <span style="color:gray;">Sample use case</span>
+|-
+|
+Firefox accounts store user data:
+
+* What happens if that data is exposed to the world? What happens if the systems go down? What happens if that data is disclosed? What if the data is modified?
+* Are we at financial, operational risk? Is our reputation also at risk?
+* What kind of data does Firefox accounts exactly handle? How sensitive is the primary kind of data used by Firefox accounts?
+
+The RRA table facilitate the answer to these questions.
+|}
+
+== When to run RRAs? ==
+
+RRAs are designed to be updated and re-runnable as needed. That being said, it is recommended to run the first RRA
+during the design or architecture phase of new services, when an initial data flow diagram for the service has been created.
+
+== What to run through RRAs? ==
+
+RRAs can only be used for services. Any project belongs to or provides a service. Any component, piece of code or data
+belongs to a service. Find which service it belongs to and RRA that service, not the specific piece of code or data.
+
+Large services should be split into multiple smaller services or sub-services that handle a specific type of data and
+expose a limited set of features. This choice has to be made by the security engineer running the RRA. If the
+sub-services belong to different teams, it is a strong indicator that multiple RRAs should be ran.
+
+Large services that cannot be split up not only lead to a complex assessment, but also may indicate that the service
+itself needs to be re-designed in a more secure fashion.
+
+== What to focus on during RRAs? ==
+
+* '''Impact assessment'''. The RRA is the authority for impact levels and these are paramount.
+* '''Rationales'''. Anyone should understand why you've set a specific impact level by reading 
+* Getting '''value for the team'''. The team needs to understand what is most important to protect.
+
+What to '''NOT''' focus on:
+* Gathering security controls, figuring out how effective they are, etc. Don't do that! This information may be recorded if it comes up but do not focus on it.
+* Likelihood, Security provided by service. Don't spent much time there! These are "10, 000 feet approximations" and part of a larger risk calculation mechanism.
+
```

Comment:
One other thing that could be worth mentioning here is how we factor in planned efforts for remediation for a service; this seems to come up all the time. Guidance would include how to deal with in-flight initiatives, where they are factored in and where they are not included.